frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -231,12 +229,30 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
         parcelId !== cachedParcelDetail?.id &&
         parcelId !== activePropertyDetail?.parcelDetail?.id
       ) {
+        addContext(SidebarContextType.LOADING);
         dispatch(fetchParcelDetail({ id: parcelId }));
       } else if (
         parcelId === activePropertyDetail?.parcelDetail?.id &&
         activePropertyDetail?.propertyTypeId === 0
       ) {
-        setCachedParcelDetail(activePropertyDetail?.parcelDetail);
+        setCachedParcelDetail({
+          ...activePropertyDetail?.parcelDetail,
+          financials: getMergedFinancials([
+            ...(activePropertyDetail?.parcelDetail?.evaluations ?? []),
+            ...(activePropertyDetail?.parcelDetail?.fiscals ?? []),
+          ]),
+        });
+        if (!!activePropertyDetail?.parcelDetail?.buildings?.length) {
```

Comment:
logic to handle whether to display view or update, and whether this is bare or developed land.